### ISSUE TITLE
HTML5 Validation not showing

### ISSUE THREAD

**Issue by markgoodyear** [adamcoulombe/jquery.customSelect#19]
I'm implementing this on a form with HTML5 validation, however once the plugin is activated, the selects which have validation do not show the browsers "tooltip" to show it's a required field. I'm guessing this is to do with the opacity.

Is there any way around this?

Cheers
Mark

**Comment by adamcoulombe** [comment]
I don't get a tooltip on a regular select box even in chrome.

http://jsfiddle.net/wAVvh/

Could you post an example?

**Comment by markgoodyear** [comment]
Sure, you just needed the submit button to submit the form (validation only occurs when you submit):

http://jsfiddle.net/wAVvh/1/

**Comment by adamcoulombe** [comment]
Ah, I see what you mean.

It is indeed the opacity issue you mentioned. I don't have an out-of-the box solution. We can probably built in something to listen for an invalid event or check for an invalid pseudo class or something. And then you can use :after content to display a custom message like so:

http://jsfiddle.net/adamco/2Kcbp/1/

**Comment by markgoodyear** [comment]
This could work nicely, though it'd need other errors to be styled in the same way otherwise there would be consistency issues — probably not something this plugin would handle though.

**Comment by adamcoulombe** [comment]
Yeah.. you can restyle your message bubbles, but only in webkit for now http://jsfiddle.net/adamco/CJBAR/1/

Just curious, if you're going with HTML5 validation why not also just style your select boxes with pure css3?

http://jsfiddle.net/adamco/BaV96/  no plugin or js needed

**Comment by markgoodyear** [comment]
It looks like changing `opacity:0` to using `appearance:none` does the trick as in the above fiddle (http://jsfiddle.net/adamco/BaV96/).

This could be used with some feature detection to fallback to `opacity:0` as it's not supported in IE or Opera, but still not a solid solution.

**Comment by adamcoulombe** [comment]
something like this may work for you...
1. wrap in label (or some other container)
2. make label `overflow:hidden`
3. style label as if it were the select box. give it the width you want your box to be
4. make select box `appearance:none`; `backgound:none`
5. make select box wider than container label so that the arrow box is cut off

http://jsfiddle.net/adamco/Xbk2r/1/

**Comment by markgoodyear** [comment]
I'll check that out. Do you know if it works on IE8+?

**Comment by adamcoulombe** [comment]
It should work in IE8.

Anyway, I'm going to leave this issue open in case Anyone has any ideas to resolve this
